File: src/main/java/Hand.java
package main.java;

import java.util.ArrayList;

public class Hand {

    private final ArrayList<Card> hand;
    private int currentScore;
    private final String name;
    private boolean hasAce;

    public String getName() {
        return name;
    }

    public Hand(String name){
        this.name = name;
        this.hand = new ArrayList<>();
        currentScore = 0;
    }

    public void addToHand(Card card){
        hand.add(card);
        if(card.isAce()) {
            hasAce = true;
        }
        currentScore += card.getCardValue();
    }

    public String showCards(){
        return this.name + ": " + String.join(", ",hand.toString().replace("[", "").replace("]", ""));
    }

    public boolean hasAce(){
        return hasAce;
    }

    public int getCurrentScore(){
        return currentScore;
    }
}

File: src/main/kotlin/Hand.kt
package main.kotlin

import java.util.ArrayList

class Hand(val name: String) {
    private val hand: ArrayList<Card> = ArrayList()
    private var hasAce = false

    fun addToHand(card: Card) {
        hand.add(card)
        if (card.isAce) {
            hasAce = true
        }
    }

    fun getScore(): Int {
        var score = 0
        for (card in hand) {
            score += card.getCardValue()
        }
        return score
    }

    fun showCards(): String {
        return name + ": " + java.lang.String.join(", ", hand.toString().replace("[", "").replace("]", ""))
    }

    fun hasAce(): Boolean {
        return hasAce
    }

}



